editor/project_manager.cpp

```diff
@@ -1326,26 +1336,21 @@ void ProjectManager::_load_recent_projects() {
 		String path = item.path;
 		String conf = item.conf;
 
-		Ref<ConfigFile> cf = memnew(ConfigFile);
-		Error cf_err = cf->load(conf);
+		if (filter_option == ProjectListFilter::FILTER_NAME && search_term != "" && item.project_name.findn(search_term) == -1)
+			continue;
 
-		String project_name = TTR("Unnamed Project");
-		if (cf_err == OK && cf->has_section_key("application", "config/name")) {
-			project_name = static_cast<String>(cf->get_value("application", "config/name")).xml_unescape();
+		if (item.config_version > ProjectSettings::CONFIG_VERSION) {
+			// Comes from an incompatible (more recent) Godot version, grey it out
+			item.grayed = true;
```

Comment:
Note: most of the code in this area looks to be re-organized, but this part seems new. Something to look over before merging.